pybamm/experiments/experiment.py

```diff
@@ -440,3 +454,16 @@ def is_cccv(self, step, next_step):
             if op["events"] == {k: v for k, v in next_op.items() if k in op["events"]}:
                 return True
         return False
+
+    def search_tag(self, tag):
```

Comment:
seems like this should use the "tags" entry of the dictionary?

Reply:
That was my original idea, but then I couldn't find a clear mapping between `operating_conditions` and the actual cycle number. Going forward, when we implement #2322, that will have to be addressed and will make more sense to use the dictionary. But for the time being, processing the strings is fast enough.